IdentityServer/v6/docs/content/fundamentals/claims.md

```diff
@@ -7,63 +7,77 @@ weight: 45
 IdentityServer emits claims about users and clients into tokens. You are in full control of which claims you want to emit, in which situations you want to emit those claims, and where to retrieve those claims from.
 
 ## User claims
-User claims can be put in both identity and access tokens. The central extensibility point to implement for emitting claims is called the [profile service]({{< ref "/reference/services/profile_service" >}}).
+User claims can be emitted in both identity and access tokens and in the [userinfo endpoint]({{< ref "/reference/endpoints/userinfo" >}}). The central extensibility point to implement to emit claims is called the [profile service]({{< ref "/reference/services/profile_service" >}}). The profile service is responsible for both gathering claim data and deciding which claims should be emitted.
 
-Whenever IdentityServer creates tokens for a user, it invokes the registered profile service with a [context]({{< ref "/reference/services/profile_service#duendeidentityservermodelsprofiledatarequestcontext" >}}) that presents detailed information about the current token request, including
+Whenever IdentityServer needs the claims for a user, it invokes the registered profile service with a [context]({{< ref "/reference/services/profile_service#duendeidentityservermodelsprofiledatarequestcontext" >}}) that presents detailed information about the current request, including
 
-* the identity of the client who is requesting the token
+* the client that is making the request
 * the identity of the user
-* what type of token is requested
-* the requested claim types according to the definition of the requested resources
+* the type of the request (access token, id token, or userinfo)
+* the requested claim types, which are the claims types associated with requested identity resource scopes
```

Comment:
not all requests to the profile service are for claim types defined by "IdentityResources", right? they could be claim types defined by ApiScopes or ApiResources too.

Reply:
Thanks for catching that! Fixed